core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpSigner.java

```diff
@@ -15,31 +15,62 @@
 
 package software.amazon.awssdk.http.auth.spi;
 
-import java.nio.ByteBuffer;
-import org.reactivestreams.Publisher;
+import java.util.function.Consumer;
 import software.amazon.awssdk.annotations.SdkPublicApi;
-import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultAsyncHttpSignRequest;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultSyncHttpSignRequest;
+import software.amazon.awssdk.identity.spi.Identity;
 
 /**
  * Interface for the process of modifying a request destined for a service so that the service can authenticate the SDK
- * customer’s identity
+ * customer’s identity.
+ *
+ * @param <IdentityT> The type of the identity.
  */
 @SdkPublicApi
-public interface HttpSigner {
+public interface HttpSigner<IdentityT extends Identity> {
+
+    /**
+     * Method that takes in inputs to sign a request with sync payload and returns a signed version of the request.
+     *
+     * @param request The inputs to sign a request.
+     * @return A signed version of the request.
+     */
+    SyncSignedHttpRequest sign(SyncHttpSignRequest<IdentityT> request);
```

Comment:
`? extends IdentityT`? (to allow someone passing a `SyncHttpSignRequest<AwsSessionIdentity>` to a `HttpSigner<AwsIdentity>`)

Reply:
Oh that's a bad miss! Thanks.